src/Microsoft.OData.Edm/ExtensionMethods/AnnotationsTargetHelper.cs

```diff
@@ -0,0 +1,205 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="AnnotationsTargetHelper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    using System.Collections.Generic;
+    using Microsoft.OData.Edm.Vocabularies;
+
+    /// <summary>
+    /// AnnotationsTarget helper
+    /// </summary>
+    internal static class AnnotationsTargetHelper
+    {
+        public static IEdmVocabularyAnnotatable HandleTargetSegments(IEdmModel model, List<IEdmElement> results, string[] targetSegments, int targetSegmentsCount, int level)

```

Comment:
Document this method. That way it's possible to know what the method does, and what `List<IEdmElement> results` parameter represents - including whether `results` is the most appropriate as parameter name, and why it's necessary to pass a `string[] targetSegments` parameter as well as an `int targetSegmentsCount` parameter. And what `int level` parameter represents